### PR TITLE
Obsolete StringHelper.Replace

### PR DESCRIPTION
**Pull request by hazzik**
`StringHelper.Replace*` methods are highly ineffective when the input string has many occurrences of a placeholder.

### CODE REVIEW

File: src/NHibernate/Impl/AbstractQueryImpl.cs [hunk 1]
```diff
@@ -265,7 +266,10 @@ private string ExpandParameterList(string query, string name, TypedValue typedLi
 
 			var paramPrefix = isJpaPositionalParam ? StringHelper.SqlParameter : ParserHelper.HqlVariablePrefix;
 
-			return StringHelper.Replace(query, paramPrefix + name, string.Join(StringHelper.CommaSpace, aliases), true);
+			return Regex.Replace(
+				query,
+				$@"{paramPrefix}{name}\b",
```
Comment (bahusoid) [inline src/NHibernate/Impl/AbstractQueryImpl.cs]: Shouldn't this place and others be wrapped in  `Regex.Escape` like `Regex.Escape(paramPrefix+name)`?

Comment (hazzik) [reply src/NHibernate/Impl/AbstractQueryImpl.cs]: This is unlikely that parameter name will contain any special regex characters.

File: src/NHibernate/Util/StringHelper.cs [hunk 2]
```diff
@@ -83,11 +83,15 @@ public static string Repeat(string str, int times)
 			return buf.ToString();
 		}
 
+		//TODO: Inline
 		public static string Replace(string template, string placeholder, string replacement)
```
Comment (fredericDelaporte) [inline src/NHibernate/Util/StringHelper.cs]: Why not obsoleting and inlining it right now?

Comment (hazzik) [reply src/NHibernate/Util/StringHelper.cs]: Did not want to change much of the code.

Comment (fredericDelaporte) [reply src/NHibernate/Util/StringHelper.cs]: Well, if not now, why later? Better even remove the TODO then.

Comment (hazzik) [reply src/NHibernate/Util/StringHelper.cs]: TODO removed

**Pull request merged**